src/Microsoft.AspNetCore.Identity/IdentityOptions.cs

```diff
@@ -11,6 +13,18 @@ namespace Microsoft.AspNetCore.Builder
     /// </summary>
     public class IdentityOptions
     {
+        public IdentityOptions(IEnumerable<IConfigureOptions<IdentityOptions>> configureOptions = null)
```

Comment:
It actually will be pretty much the same, you'll need to register your option explicitly is the only difference:

```
services.AddOptions<AppOptions>();
services.Configure<AppOptions>(appOptions =>
{
    appOptions.Key = "value";
    ...
});
```

You won't need to change your AppOptions if you use this approach.

If you want to avoid using AddOptions, you can do the alternative
```
// AppOptions must take the IEnumerable<IConfigureOptions<AppOptions>> 
// and apply them in the ctor if you choose this route
services.AddSingleton<AppOptions>();
services.Configure<AppOptions>(appOptions =>
{
    appOptions.Key = "value";
    ...
});
```